server/src/main/java/org/opensearch/plugins/SearchPlugin.java

```diff
@@ -138,6 +140,14 @@ default Map<String, Highlighter> getHighlighters() {
         return emptyMap();
     }
 
+    /**
+     * Allows plugins to register custom decider for concurrent search
+     * @return A {@link Collection} of {@link ConcurrentSearchDecider}
+     */
+    default Collection<ConcurrentSearchDecider> getConcurrentSearchDeciders() {
+        return emptyList();
```

Comment:
Sorry I missed this in previous rounds of review. But why we want plugins to return list of deciders ? I think single decider should be good here.

Reply:
Also lets annotate this as experimental as well

Reply:
I thought that if plugins want to add additional deciders based on plugin specific logic, they can do so. But yeah, probably it's not necessary